features/step_definitions/networking.rb

```diff
@@ -1437,3 +1437,21 @@
   end
   logger.info "The openfolw egressip related table number #{cb[cb_name]} is stored to the #{cb_name} clipboard."
 end
+
+Given /^I switch the ovn gateway mode on this cluster$/ do
+  ensure_admin_tagged
+  step "I store the masters in the clipboard"
+  ovnkube_master = BushSlicer::Pod.get_labeled("app=ovnkube-master", project: project("openshift-ovn-kubernetes", switch: false), user: admin, quiet: true) { |pod, hash| pod.node_name == node.name}.first
+  @result = admin.cli_exec(:logs, resource_name: ovnkube_master.name, n: "openshift-ovn-kubernetes", c: "ovnkube-master")
+  
+  if @result[:response].include? "Gateway:{Mode:local"
+    logger.info "OVN Gateway mode is Local. Changing Gateway mode to Shared now..."
+    @result = admin.cli_exec(:patch, resource: "network.operator", resource_name: "cluster", p: "{\"spec\":{\"defaultNetwork\":{\"ovnKubernetesConfig\":{\"gatewayConfig\":{\"routingViaHost\": false}}}}}", type: "merge")
```

Comment:
How do we intend to track progress of the network operator? In the step definition or in the scripts?

Reply:
@asood-rh ```oc rollout status``` in later step will ensure that network operator has uodated daemonsets properly and rolled out the changes successfully.

Reply:
Shall we need check network co in correct status ("oc get co network") before and after switch gateway mode to make sure at least switching won't cause network co wrong ?

Reply:
@huiran0826 If ```oc rollout``` operation fails it will leave the operator in unexpected state. But sure it would be helpful in adding extra check as well as @asood-rh also mentioned. Let me add pre-check/post check for operator post in scenario itself for more clarity. Thanks guys.

Reply:
Done and re-ran https://privatebin.corp.redhat.com/?9735a6ed58358bf8#5kg1NTq7QmAV9qcbkhcpSb2eGhy1zEpewBhe6iwhPttk